### PR TITLE
leetcode-cli: init at 0.3.10

### PR DESCRIPTION
**Pull request by Congee**
###### Motivation for this change
This tool make using leetcode.com on the command line possible. It is analogous to `pkgs/applications/misc/exercism`. It would be nice to include this package in nixpkgs so users wouldn't have to built it themselves.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2382,6 +2382,13 @@
     githubId = 11145016;
     name = "J.C.";
   };
+  congee = {
```
Comment (cole-h) [inline maintainers/maintainer-list.nix:2385]: Would you mind making a separate commit for adding yourself as a maintainer (before the `leetcode: init` commit, git history-wise), with the message resembling `maintainers: add congee`? Thanks!

Comment (Congee) [reply maintainers/maintainer-list.nix:2385]: sure and thank you

Comment (Congee) [reply maintainers/maintainer-list.nix:2385]: just did ;P

Comment (cole-h) [reply maintainers/maintainer-list.nix:2385]: Thank you!

**Review by cole-h — APPROVED** [review]
LGTM, thanks! I verified that the CLI seemed to do things (but I don't use leetcode, so I couldn't go much further than "does `leetcode -h` do stuff").

**Comment by Congee** [comment]
Thanks a lot Cole. Yeah it doesn't do much without an account

**Pull request merged**